### PR TITLE
Expand k8s workload selectors

### LINKED ISSUE #715: Update the Kubernetes workload attestor
The k8s workload attestor currently only emits namespace and service account selectors, but we have access to much more than that. Update the attestor to emit additional information. At a minimum, the following should be emitted in addition to service account and namespace:
* pod labels
* container name
* container image
* node name

### PR DESCRIPTION
**Pull request by azdagron**
Adds support for the following additional selectors:
- pod UID (e.g. `pod-uid:012345678-0123-0123-0123-0123456789abc`)
- pod labels by name and value (e.g. `pod-label:version:v0`)
- owner reference by kind and name, (e.g. `pod-owner:DaemonSet:foo`)
- container image (e.g. `container-image:name:latest`)
- container name (e.g. `container-name:blog`)
- node name (e.g. `node-name:k8s-node-1`)

In addition, refactored the unit-tests into a test suite to reduce code duplication and improved test case readability.

Fixes #715

### CODE REVIEW

File: pkg/agent/plugin/workloadattestor/k8s/k8s.go [hunk 1]
```diff
@@ -52,10 +52,21 @@ type k8sPluginConfig struct {
 type podInfo struct {
 	// We only care about namespace, serviceAccountName and containerID
 	Metadata struct {
-		Namespace string `json:"namespace"`
+		UID             string            `json:"uid"`
+		Namespace       string            `json:"namespace"`
+		Labels          map[string]string `json:"labels"`
+		OwnerReferences []struct {
+			Kind string `json:"kind"`
+			Name string `json:"name"`
```
Comment (mcpherrinm) [inline pkg/agent/plugin/workloadattestor/k8s/k8s.go]: Might be nice to have the uid here too, as an alternative to the name

Comment (azdagron) [reply pkg/agent/plugin/workloadattestor/k8s/k8s.go]: We'd need distinct selectors for name v.s. uid. Curious on what formats people think are resonable. Maybe something like:

```
pod-owner-name:DaemonSet:blog
pod-owner-uid:DaemonSet:01234567-0123-0123-0123-0123456789abc
```

Alternatively, we could parse the owner reference Kind (i.e. `DaemonSet` and `ReplicationController`) and produce:

```
daemon-set-name:blog
daemon-set-uid:01234567-0123-0123-0123-0123456789abc
```
or

```
replica-set-name:blog
replica-set-uid:01234567-0123-0123-0123-0123456789abc
```

The only downside i can see to the latter approach is having to extend the plugin to recognize new OwnerReference kind's.

Comment (mcpherrinm) [reply pkg/agent/plugin/workloadattestor/k8s/k8s.go]: pod-owner:DaemonSet:blog
pod-owner-uid:DaemonSet:01234567-0123-0123-0123-0123456789abc

IMO

Comment (azdagron) [reply pkg/agent/plugin/workloadattestor/k8s/k8s.go]: We could also construct the name using something like https://godoc.org/github.com/gobuffalo/flect#Dasherize to mitigate the concern over new OwnerReference kinds.

Comment (azdagron) [reply pkg/agent/plugin/workloadattestor/k8s/k8s.go]: On more thought, dasherizing the name would provide unnecessary friction for those crafting registration entries from PodSpec's (they'd have to follow the same convention).  I waffled a bit on `pod-owner` v.s. `pod-owner-name`, but I'm inclined to embrace your proposal.

File: pkg/agent/plugin/workloadattestor/k8s/k8s.go [hunk 2]
```diff
@@ -207,21 +222,52 @@ func lookUpContainerInPod(containerID string, status podStatus) containerLookup
...
+	for k, v := range info.Metadata.Labels {
+		selectors = append(selectors, makeSelector("pod-label:%s:%s", k, v))
+	}
+	for _, ownerReference := range info.Metadata.OwnerReferences {
+		selectors = append(selectors, makeSelector("pod-owner:%s:%s", ownerReference.Kind, ownerReference.Name))
+		selectors = append(selectors, makeSelector("pod-owner-uid:%s:%s", ownerReference.Kind, ownerReference.UID))
+	}
+
+	for _, container := range info.Spec.Containers {
+		if container.Name != containerName {
```
[the remark this answers is not in the document]
Comment (azdagron) [reply pkg/agent/plugin/workloadattestor/k8s/k8s.go]: From what I can tell, the container ID isn't part of the  container spec. However, both the name and image are available on the container status... maybe I'll just return those properties from lookUpContainerInPod and bypass this container spec search altogether?

**Pull request merged**